cmd/minecraft-preempt/proxy.go

```diff
@@ -159,17 +159,21 @@ func (p *Proxy) Start(ctx context.Context) error {
 		}
 
 		// create a new connection
-		conn := NewConnection(&rawConn, p.log, p.server)
+		conn := NewConnection(&rawConn, p.log, p.server, &ConnectionHooks{
+			OnLogin: func() {
+				// reset the emptySince time
+				p.emptySince.Store(nil)
+				p.connections.Add(1)
+			},
+			OnClose: func() {
+				p.connections.Add(^uint64(0))
```

Comment:
This is going to trigger even on invalid connections. I need to do some state tracking here for this logic.